runelite-client/src/test/java/net/runelite/client/plugins/npchighlight/NpcIndicatorsPluginTest.java

```diff
@@ -81,4 +85,148 @@ public void getHighlights()
+		when(npcIndicatorsConfig.highlightMenuNames()).thenReturn(false);
+		final MenuEntry menuEntry = mock(MenuEntry.class);
+		when(menuEntry.getTarget()).thenReturn("<col=ffff00>Tekton");
+		final NPC npc = mock(NPC.class);
+		when(npc.isDead()).thenReturn(false);
+		when(npc.getName()).thenReturn("Tekton");
+		when(npc.getCombatLevel()).thenReturn(0);
+
+		String recoloredTarget = npcIndicatorsPlugin.buildRecoloredNpcMenuEntryTarget(menuEntry, npc);
+		assertEquals("<col=ffff00>Tekton", recoloredTarget);
+	}
+
+	@Test
+	public void buildRecoloredNpcMenuEntryTarget_withoutCombatLevel_dead_noHighlight()
+	{
+		npcIndicatorsPlugin.highlightedNpcs.clear();
+		when(npcIndicatorsConfig.dampenDeadNPCs()).thenReturn(true);
+		when(npcIndicatorsConfig.highlightMenuNames()).thenReturn(false);
+		final MenuEntry menuEntry = mock(MenuEntry.class);
+		when(menuEntry.getTarget()).thenReturn("<col=ffff00>Tekton");
+		final NPC npc = mock(NPC.class);
+		when(npc.isDead()).thenReturn(true);
+		when(npc.getName()).thenReturn("Tekton");
+		when(npc.getCombatLevel()).thenReturn(0);
+
+		String recoloredTarget = npcIndicatorsPlugin.buildRecoloredNpcMenuEntryTarget(menuEntry, npc);
+		assertEquals("<col=b2b200>Tekton", recoloredTarget);
+	}
+
+	@Test
+	public void buildRecoloredNpcMenuEntryTarget_withoutCombatLevel_alive_highlight()
+	{
+		npcIndicatorsPlugin.highlightedNpcs.clear();
+		when(npcIndicatorsConfig.dampenDeadNPCs()).thenReturn(true);
+		when(npcIndicatorsConfig.highlightMenuNames()).thenReturn(true);
+		when(npcIndicatorsConfig.getHighlightColor()).thenReturn(new Color(255, 200, 255));
+		final MenuEntry menuEntry = mock(MenuEntry.class);
+		when(menuEntry.getTarget()).thenReturn("<col=ffff00>Tekton");
+		final NPC npc = mock(NPC.class);
+		when(npc.isDead()).thenReturn(false);
+		when(npc.getName()).thenReturn("Tekton");
+		when(npc.getCombatLevel()).thenReturn(0);
+		npcIndicatorsPlugin.highlightedNpcs.add(npc);
+
+		String recoloredTarget = npcIndicatorsPlugin.buildRecoloredNpcMenuEntryTarget(menuEntry, npc);
+		assertEquals("<col=ffc8ff>Tekton", recoloredTarget);
+	}
+
+	@Test
+	public void buildRecoloredNpcMenuEntryTarget_withoutCombatLevel_dead_highlight()
+	{
+		npcIndicatorsPlugin.highlightedNpcs.clear();
+		when(npcIndicatorsConfig.dampenDeadNPCs()).thenReturn(true);
+		when(npcIndicatorsConfig.highlightMenuNames()).thenReturn(true);
+		when(npcIndicatorsConfig.getHighlightColor()).thenReturn(new Color(255, 200, 255));
+		final MenuEntry menuEntry = mock(MenuEntry.class);
+		when(menuEntry.getTarget()).thenReturn("<col=ffff00>Tekton");
+		final NPC npc = mock(NPC.class);
+		when(npc.isDead()).thenReturn(true);
+		when(npc.getName()).thenReturn("Tekton");
+		when(npc.getCombatLevel()).thenReturn(0);
+		npcIndicatorsPlugin.highlightedNpcs.add(npc);
+
+		String recoloredTarget = npcIndicatorsPlugin.buildRecoloredNpcMenuEntryTarget(menuEntry, npc);
+		assertEquals("<col=b28cb2>Tekton", recoloredTarget);
```

Comment:
I guess this should be `assertEquals(ColorUtil.prependColorTag("Tekton", Color.YELLOW.darker()), recoloredTarget);`. Same for other tests